[RFC] Fix stall before starting tui when something is output in Windows

erw7 opened the pull request:
fixes #7967

erw7 committed:
os/input.c: fixes #7967

erw7 commented:
I don't know if it's really all right, but is it better to fix `stream_init()` like https://github.com/neovim/neovim/commit/a48ea6806dac6f27accc8ca19628538111ea98f2?

justinmk commented:
> I don't know if it's really all right, but is it better to fix `stream_init()` like [6589f6f](https://github.com/neovim/neovim/commit/6589f6f418fed08fd9e9f1d063989543e9324c22)?

That seems slightly better to me, if this is something that's always required for streams on Windows.

But for #7967 I think the best approach is to simply not print anything until startup is ready, as  @bfredl  [mentioned](https://github.com/neovim/neovim/issues/7967#issuecomment-433875000). Alternatively, collect messages in a list which is later printed.

erw7 commented:
I do not understand exactly what it does, but I implemented it referring to the embedded case. The behavior is different from vim(A message is displayed after the screen is cleared, but on vim a message is displayed before the screen is cleared), but it seems to be working. This closes because I created #9829.

erw7 committed:
Revert "os/input.c: fixes #7967"

This reverts commit b860d05ceecf3ad797d832a19390f70bf42f13e6.

erw7 committed:
win: stream_init() issue with tty on Windows

erw7 committed:
main.c: Fix reading handle issue

Change to use COIN$ for reading when using stdin for command reading on
Windows.

justinmk commented on src/nvim/tui/input.c:
```diff
@@ -49,14 +49,9 @@ void tinput_init(TermInput *input, Loop *loop)
-#else
   rstream_init_fd(loop, &input->read_stream, input->in_fd, 0xfff);
+#ifdef WIN32
+  uv_tty_set_mode(&input->read_stream.uv.tty, UV_TTY_MODE_RAW);
```
why not do this in `stream_init` ?

erw7 replied:
All current calls need to do `uv_tty_set_mode(..., ..., UV_TTY_MODE_RAW)`, so I think that's fine. There may be calls that do not need it, so I did not.
